crates/server/src/store/mod.rs

```diff
@@ -67,3 +67,30 @@ pub trait StoreBackend {
     #[cfg(feature = "metrics")]
     fn metrics_label(&self) -> &'static str;
 }
+
+/// Private helper functions for working with `StoreBackend` implementations
```

Comment:
Why create a separate trait for this? Extension traits are typically used if either the the type or trait is foreign (not from the same crate), which is not the case here.